qt/python/mantidqt/mantidqt/widgets/sliceviewer/cutviewer/view.py

```diff
@@ -0,0 +1,210 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantid workbench.
+#
+from qtpy.QtWidgets import QVBoxLayout, QWidget, QTableWidget, QHeaderView, QTableWidgetItem
+from qtpy.QtCore import Qt
+from qtpy.QtGui import QColor
+from mantid.plots.plotfunctions import create_subplots
+from mantidqt.MPLwidgets import FigureCanvas
+from mantidqt.plotting.mantid_navigation_toolbar import MantidNavigationToolbar
+import matplotlib.text as text
+from mantid.simpleapi import AnalysisDataService as ADS
+from mantid.kernel import SpecialCoordinateSystem
+from numpy import zeros
+
+# local imports
+from .representation.cut_representation import CutRepresentation
+
+
+class CutViewerView(QWidget):
+    """Displays a table view of the PeaksWorkspace along with controls
+    to interact with the peaks.
+    """
+    def __init__(self, presenter, canvas, frame, parent=None):
+        """
+        :param painter: An object responsible for drawing the representation of the cut
+        :param sliceinfo_provider: An object responsible for providing access to current slice information
+        :param parent: An optional parent widget
+        """
+        super().__init__(parent)
+        self.presenter = presenter
+        self.layout = None
+        self.figure_layout = None
+        self.table = None
+        self.figure = None
+        self.cut_rep = None
+        self.canvas = canvas
+        self.frame = frame
+        self._setup_ui()
+        self._init_slice_table()
+        self.table.cellChanged.connect(self.on_cell_changed)
+
+    def hide(self):
+        super().hide()
+        if self.cut_rep is not None:
+            self.cut_rep.remove()
+            self.cut_rep = None
+
+    # signals
+
+    def on_cell_changed(self, irow, icol):
+        self.set_bin_params(*self.presenter.validate_bin_params(irow, icol))
```

Comment:
We're leaking knowledge of the presenter here - if this slot was on the presenter this could be avoided?

Reply:
Yes I should have made a slot for this, there are still other instances in the view where I call a method on the presenter as I didn't think this too bad, but I can try and remove them.